### ISSUE TITLE
Parsing of doubles fails in certain cases

### ISSUE THREAD

**Issue by jjbr** [javaparser/javaparser#1980]
An interesting edge case this one. According to the JLS you can define numeric values with underscores at any position (for code readability). In other words, `100000.0 == 100_000.0 == 1_0_0_0_0_0.0`. https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.10.2

However, according to the documentation of e.g. `java.lang.Double#valueOf(java.lang.String)` (which is referenced from `Double#parseDouble(String)` to use the same format) these underscores cause a `NumberFormatException` when parsing. https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#valueOf-java.lang.String-

In other words, it is not possible to blindly trust methods such as `Double#parseDouble` to parse any double as specified in Java source code.

**Update: I just found that we already consider this problem in ~some~ most cases.** It only appears to be missing in the `DoubleLiteralExpr#asDouble` case. Still leaving the text above for future reference, in case someone wonders why underscores need to be replaced in the string. Will submit a fix shortly.

**Comment by HoldYourWaffle** [comment]
> An interesting edge case this one. According to the JLS you can define numeric values with underscores at any position (for code readability). In other words, 100000.0 == 100_000.0 == 1_0_0_0_0_0.0. 

This project never stops amazing me with weird syntaxes.

**Comment by matozoid** [comment]
@jjbr thanks, I'm just so confused, why didn't they implement this correctly? :-/ Anyway, great issue!

@HoldYourWaffle - my favourites are array brackets just before the opening { of a method, and hexadecimal floating point literals.
